### ISSUE TITLE
Uncaught TypeError: Cannot read property 'iterator' of undefined

### ISSUE THREAD

**Issue by EasyLian** [parse-community/Parse-SDK-JS#916]
parse.weapp.js  2.7.1    (parse.weapp.min.js)

Such a problem has arisen.

unavailable

I don't know if it's just me.

```
Uncaught TypeError: Cannot read property 'iterator' of undefined
    at module.exports (parse.weapp.js:22990)
    at Object.256.../internals/has (parse.weapp.js:22082)
    at o (parse.weapp.js:7)
    at parse.weapp.js:7
    at Object.219.../internals/create-property-descriptor (parse.weapp.js:21330)
    at o (parse.weapp.js:7)
    at parse.weapp.js:7
    at Object.222.../internals/create-iterator-constructor (parse.weapp.js:21371)
    at o (parse.weapp.js:7)
    at parse.weapp.js:7
```

**Comment by davimacedo** [comment]
Can you please share the code that is throwing this error?

**Comment by EasyLian** [comment]
> Can you please share the code that is throwing this error?
Ordinary require
Nothing else was done.

WeChat Developer Tool : https://developers.weixin.qq.com/miniprogram/en/dev/devtools/devtools.html

`const Parse = require("../../utils/parse.weapp.js")`

**Comment by EasyLian** [comment]
@davimacedo
> Can you please share the code that is throwing this error?


Ordinary require
Nothing else was done.

WeChat Developer Tool : https://developers.weixin.qq.com/miniprogram/en/dev/devtools/devtools.html

`const Parse = require("../../utils/parse.weapp.js")`

**Comment by davimacedo** [comment]
@einsqing do you mind to take a look at this issue?

**Comment by EasyLian** [comment]
I tested 3 versions of 2.71 2.70 2.60, 2.60 had no problem, only 2.71 2.70 had problems.

**Comment by dplewis** [comment]
The only change I see that could cause this would be the removing of babel/polyfill https://github.com/parse-community/Parse-SDK-JS/pull/877.

**Comment by EasyLian** [comment]
Tested.
It seems that this is not the problem.

**Comment by dplewis** [comment]
https://github.com/parse-community/Parse-SDK-JS/compare/2.6.0...2.7.0

The other  maybe cause  is a corejs3 package update

**Comment by EasyLian** [comment]
I searched the corresponding key code of the project once, and according to my survey, it was supposed to add the question of "core-js": "3.2.1".

**Comment by davimacedo** [comment]
Do you mean that it is missing? It is actually in our package.json: https://github.com/parse-community/Parse-SDK-JS/blob/master/package.json#L54

**Comment by dplewis** [comment]
@EasyLian did installing that fix your issue? If so we would have to move it from devdependency

**Comment by EasyLian** [comment]
No, I mean the problem caused by adding this core. js. I don't know how to solve it.

**Comment by davimacedo** [comment]
Have you tested without it?

**Comment by dplewis** [comment]
We could revert https://github.com/parse-community/Parse-SDK-JS/pull/877 as it introduce corejs.. You said this wasn’t the issue but then it is the issue.

Unfortunately I can’t signup for WeChat Mini Program as I don’t know Chinese or have friends to ask for an account to replicate this issue.

@einsqing Can you help us resolve this?

**Comment by einsqing** [comment]
Okay, I'll have a look when I'm free

**Comment by einsqing** [comment]
@EasyLian 没找到原因 @dplewis I can provide you with a wechat mini program, but I need you to provide a wechat number.

**Comment by cshlxm** [comment]
have the same problem  on v2.8.0 ,have not been solved yet

**Comment by einsqing** [comment]
@cshlxm No, I've been using version 2.6.0

**Comment by cshlxm** [comment]
> @cshlxm No, I've been using version 2.6.0

V2.6.0  have no problem,  I have tested .

**Comment by dplewis** [comment]
Can one of you check if reverting the changes made https://github.com/parse-community/Parse-SDK-JS/pull/877 fixes this issue for you. Or maybe there is a babel option that would work.

**Comment by cshlxm** [comment]
> Can one of you check if reverting the changes made #877 fixes this issue for you. Or maybe there is a babel option that would work.
 @dplewis    I have checked commit 81ddc94  ,the Error still exist. And I  also checked  commit  2f809ead  ,there are no Error any more.

**Comment by dplewis** [comment]
@cshlxm By revert I mean remove corejs3 and re-add babel polyfill and see if that works so we know if corejs3 is the problem or not.

**Comment by cshlxm** [comment]
@dplewis   It works.  I just remove corejs3,and re-add babel polyfill .And remove what modified in gulpfile.js and babel-jest-js. Rebuild it as V2.8.0 , and It works in WeChat Mini Program .

**Comment by dplewis** [comment]
I knew it. I really don’t to revert this change as babel-polyfill is deprecated.

Is there a corejs3 setting that could fix this issue?

I’m not sure if we can have both. But you are more than welcome to try. 

Worst case is one of you fork the project for wechat and keep it up to date. It won’t be managed by the community until corejs has a fix or somebody finds a fix.

@acinader Thoughts?

**Comment by dplewis** [comment]
Closing via https://github.com/parse-community/Parse-SDK-JS/issues/1127
